@shared/api/internal.ts

```diff
@@ -467,6 +481,156 @@ export const getAccountIndexerBalances = async (
   }
 };
 
+const getSorobanTokenBalance = async (
+  server: SorobanRpc.Server,
+  contractId: string,
+  txBuilders: {
+    // need a builder per operation, Soroban currently has single op transactions
+    balance: TransactionBuilder;
+    name: TransactionBuilder;
+    decimals: TransactionBuilder;
+    symbol: TransactionBuilder;
+  },
+  balanceParams: xdr.ScVal[],
+) => {
+  // Right now we can only have 1 operation per TX in Soroban
+  // for now we need to do 4 tx simulations to show 1 user balance. :(
+  // TODO: figure out how to fetch ledger keys to do this more efficiently
+  const decimals = await getDecimals(contractId, server, txBuilders.decimals);
+  const name = await getName(contractId, server, txBuilders.name);
+  const symbol = await getSymbol(contractId, server, txBuilders.symbol);
+  const balance = await getBalance(
+    contractId,
+    balanceParams,
+    server,
+    txBuilders.balance,
+  );
+
+  return {
+    balance,
+    decimals,
+    name,
+    symbol,
+  };
+};
+
+export const getAccountBalancesStandalone = async ({
```

Comment:
creating a standalone fetch for balances that combines our old logic of fetching classic assets from a Horizon endpoint and token balances from a Soroban RPC endpoint